router.go

```diff
@@ -160,6 +161,16 @@ func treePath(method, path string) string {
 	return "/" + method + path
 }
 
+//validMethod is copied from net/http/request.go
```

Comment:
Can we add more doc/context here? I realize the original method you cited doesn't have a doc block, but it does at least have an internal comment block that shows what it's testing for.

See https://github.com/golang/go/blob/a5d61be040ed20b5774bff1b6b578c6d393ab332/src/net/http/request.go#L817

Reply:
Improved the comment